packages/e2e/test/web-extension/extension/background/services.ts

```diff
@@ -36,6 +38,28 @@ exposeApi(
   { logger, runtime }
 );
 
+// Dummy exposeApi object that closes the port as soon as it gets a message.
+// UI promise call should reject as a result of this
+// UI consumes API -> BG exposes fake API that closes port
+runtime.onConnect.addListener((port) => {
```

Comment:
Previously adding multiple `onConnect` listeners didn't work reliably. Do the web-extension tests still pass with the same with similar consistency? We might want to use messenger for it - a full (even though typed as partial) port object should emitted on `connect$`.

Reply:
The web-extension e2e test suite is passing consistently.